src/Database/LSMTree/Internal/BloomFilter.hs

```diff
@@ -61,8 +61,13 @@ bloomFilterFromSBS (SBS ba') = do
     -- limit to 2^48 bits
     when (len >= 0xffff_ffff_ffff) $ Left "Too large bloomfilter"
 
+    -- make sure byte array is long enough for vector
+    let len64 = fromIntegral (div64 len)
+    when (2 + len64 > div8 (P.sizeofByteArray ba)) $
```

Comment:
Something I realised just now: if the byte array is much larger than the part of it that the bloom filter uses, then we are still keeping that unused part alive. Maybe we should check that the size of byte array is *exactly* `(2 + len64) * 8)`? Or that it is within a small bound? Or, possibly, we slice the byte array

Reply:
It's more restrictive than necessary, but the way we use it it should always be true, I think. So might make sense to add an assertion/check.

Reply:
I added a check.